src/core/server/saved_objects/saved_objects_service.ts

```diff
@@ -377,8 +379,8 @@ export class SavedObjectsService
      * HTTP server running without an Elasticsearch server being available.
      * So, when the `migrations.skip` is true, we skip migrations altogether.
      */
-    const cliArgs = this.coreContext.env.cliArgs;
-    const skipMigrations = cliArgs.optimize || this.config.migration.skip;
+    const skipMigrations =
+      this.config.migration.skip || this.coreContext.env.isDevClusterMaster || !pluginsInitialized;
```

Comment:
is this flag `isDevClusterMaster` already used when calculating `pluginsInitialized` ?  https://github.com/elastic/kibana/pull/61325/files#diff-725c498a267472754f01cff2baa3901bR110

Reply:
No there's a `--plugin.initialized` config option that is used (optimizer also specifies this flag)